Toast JS errors

himdel opened the pull request:
This is a devel mode only feature, which causes all JS errors to not only show up in the (browser) console, but also to pop up as a toast notification - hopefully more devs will notice in such cases.

Also, the toasts are [proper patternfly toasts](https://www.patternfly.org/patterns/toast-notifications/) with a set width, so they could be useable for other things eventually. (JS errors probably don't need a notification center or anything like that..)

Right now, it toasts on:

   * `window.onerror` - that is, any uncaught exception
   * `*.onerror` - failed to load image, etc.
   * `console.error` & `console.warn` - error outputs from libraries
   * `jQuery.globalEval` - eval errors from ajax requests

Screenshot:

![error](https://cloud.githubusercontent.com/assets/289743/12983149/de04530a-d0df-11e5-9173-958eede3f3cf.png)

martinpovolny commented:
@himdel : can you rebase this so that we merge it? Or is it no longer a desired solution?

himdel commented:
I think we still want it, in devel mode.. Will do..

himdel commented:
Rebased, rewritten to angular and communicating using the RxSubject code introduced in #9753 ..

Also made the wrapper disappear when there are no messages, added a clear-all button, and added some logic to prevent useles `[object Event]`, `[object Object]`, etc. messages...

himdel commented:
@martinpovolny could you please test on some more errors.. I *hope* I've made all the paths show something reasonable but..

Also, opening a VM detail in Compute > Infra > VMs shows an unspecific error now, that's caused by 

```
app/assets/images/100/configuration_profile.png:                             Adobe Photoshop Image, 100 x 100, RGBA, 4x 8-bit channels
app/assets/images/100/guest_application.png:                                 Adobe Photoshop Image, 100 x 100, RGBA, 4x 8-bit channels
```

(ie. not PNG) - @epwinchell is aware of it :)

skateman commented:
![](http://i0.kym-cdn.com/entries/icons/original/000/012/367/evilest.gif)

martinpovolny commented:
Maybe people actually start fixing the issues if it jumps out RED on the top ;-)

skateman commented:
My issue is with orange:
![screenshot from 2016-08-11 16-24-23](https://cloud.githubusercontent.com/assets/649130/17591831/18f5cf9a-5fe0-11e6-8c49-6225cc9a6cc9.png)

martinpovolny commented:
Shall I expect a PR fixing these duplicate DOM IDs later today?
